.travis.yml

```diff
@@ -1,28 +1,64 @@
+matrix:
+  include:
+    - os: linux
+      compiler: gcc
+    - os: osx
+      osx_image: xcode7.3
+      compiler: clang
+
 language: cpp
 
-compiler:
-  - gcc
+cache:
+  directories:
+    - ${HOME}/local
+    - ${HOME}/cpanm
 
 addons:
   hosts:
     - 127.0.0.1.xip.io
     - alternate.127.0.0.1.xip.io
 
 before_install:
-  # upgrade g++ and libstdc++ to build nghttp2
-  - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
-  - sudo apt-add-repository --yes ppa:smspillaz/cmake-2.8.12
-  - sudo apt-get --yes update
-  - sudo apt-get install --yes cmake cmake-data g++-4.8 libstdc++-4.8-dev php5-cgi wget
-  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
-  - $CXX --version
-  # for speed, pre-install deps installed in `before_script` section as ubuntu packages
-  - sudo apt-get install -qq cpanminus libipc-signal-perl liblist-moreutils-perl libwww-perl libio-socket-ssl-perl zlib1g-dev
+  # setup variables
+  - export INSTALL_DIR=${HOME}/local
+  - export CPANM_DIR=${HOME}/cpanm
+  - export PATH=`echo -n "$PATH" | sed -E "s#(:/usr(/local/bin|/bin):)#:${INSTALL_DIR}/bin:${CPANM_DIR}/bin\1#"`
+  - export PKG_CONFIG_PATH=${INSTALL_DIR}/lib/pkgconfig:/usr/local/opt/openssl/lib/pkgconfig
+  - export PERL5LIB=${CPANM_DIR}/lib/perl5
+  - export PERL_CPANM_OPT="--local-lib=${CPANM_DIR}"
+  - export PERL_MB_OPT="--install_base ${CPANM_DIR}"
+  - export PERL_MM_OPT="INSTALL_BASE=${CPANM_DIR}"
+  - export CACHE_VERSION_DIR=${INSTALL_DIR}/versions
+  - if [[ ! -d "${CACHE_VERSION_DIR}" ]]; then mkdir -p "${CACHE_VERSION_DIR}"; fi
+  # get latest version of cpanminus
+  - |
+    if [[ ! -e "${INSTALL_DIR}/bin/cpanm" ]]; then
+      curl -L https://cpanmin.us/ --create-dirs -o ${INSTALL_DIR}/bin/cpanm
+      chmod +x ${INSTALL_DIR}/bin/cpanm
+    fi
+  - |
+    case "${TRAVIS_OS_NAME}" in
+      "linux" )
+        # upgrade g++ and libstdc++ to build nghttp2
+        sudo apt-add-repository --yes ppa:ubuntu-toolchain-r/test
+        sudo apt-add-repository --yes ppa:smspillaz/cmake-2.8.12
+        sudo apt-get -qq update
+        sudo apt-get install -qq cmake cmake-data g++-4.8 libstdc++-4.8-dev php5-cgi wget
+        if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
+        $CXX --version
+        # for speed, pre-install deps installed in `before_script` section as ubuntu packages
+        sudo apt-get install -qq libipc-signal-perl liblist-moreutils-perl libwww-perl libio-socket-ssl-perl zlib1g-dev
+        ;;
+      "osx"   )
+        brew update
+        brew tap homebrew/dupes
+        brew install --quiet openssl homebrew/dupes/zlib
+        brew link --force openssl
```

Comment:
This won't work, FYI. It can cause security issues so it's blocked off.

You should either pass flags or prepend your `PATH` with the directory - Potentially `CMAKE_PREFIX_PATH` rather than `PATH` given `h2o` uses CMake.